src/nbl/video/CVulkanCommandBuffer.h

```diff
@@ -3,28 +3,435 @@
 
 #include "nbl/video/IGPUCommandBuffer.h"
 
+// Todo(achal): I think I a lot of them could be made forward declarations if I introduce
+// a CVulkanCommandBuffer.cpp
+#include "nbl/video/CVulkanBuffer.h"
+#include "nbl/video/CVulkanImage.h"
+#include "nbl/video/CVulkanComputePipeline.h"
+#include "nbl/video/CVulkanPipelineLayout.h"
+#include "nbl/video/CVulkanDescriptorSet.h"
+
 #include <volk.h>
 
-namespace nbl
-{
-namespace video
+namespace nbl::video
 {
+struct ArgumentReferenceSegment;
 
-class CVulkanCommandBuffer : public virtual IGPUCommandBuffer
+class CVulkanCommandBuffer : public IGPUCommandBuffer
 {
 public:
-    VkCommandBuffer getInternalObject() const { return m_cmdbuf; }
+    CVulkanCommandBuffer(core::smart_refctd_ptr<ILogicalDevice>&& logicalDevice, E_LEVEL level,
+        VkCommandBuffer _vkcmdbuf, IGPUCommandPool* commandPool)
```

Comment:
why isnt it  `core::smart_refctd_ptr<IGPUCommandPool>&&` ?

Reply:
@devshgraphicsprogramming can you explain when to use and not use `core::smart_refctd_ptr` 

And if you already did, can you point me to the discussion.

Reply:
done in discord.